src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala

```diff
@@ -441,13 +416,10 @@ abstract class ClassfileParser {
     }
 
     val isTopLevel = !(currentClass containsChar '$') // Java class name; *don't* try to to use Scala name decoding (SI-7532)
-
-    val c = if (isTopLevel) pool.getClassSymbol(nameIdx) else clazz
     if (isTopLevel) {
-      if (c != clazz) {
-        if ((clazz eq NoSymbol) && (c ne NoSymbol)) clazz = c
```

Comment:
Is there any scenario in e.g. presentation/interactive mode where we needed this? /cc @dragos